### ISSUE TITLE
Serve json-schema for resources

### ISSUE THREAD

**Issue by ConnorDoyle** [mesosphere/marathon#1185]
See [json-schema.org](http://json-schema.org).

Especially important are:

- [ ] AppDefinition
- [ ] AppUpdate
- [ ] Group
- [ ] DeploymentStep
- [ ] Deployment
- [ ] Timestamp

**Comment by ConnorDoyle** [comment]
Let's think about adding a new `/v2/schemas` collection.  A client could fetch the schema for AppDefinition via `GET /v2/schemas/app`, for example.

**Comment by ConnorDoyle** [comment]
I've written up a json-schema document for the current Marathon app resource to use as a starting point: https://raw.githubusercontent.com/mesosphere/universe/master/repo/meta/schema/marathon-schema.json

**Comment by drexin** [comment]
Cool, thanks! We should also use the schema to verify the output of the rest endpoints in our tests.

**Comment by sttts** [comment]
@drexin I like that idea. This also removes my complaint that the schema is in parallel to the Reader and Writers for the api.

**Comment by drexin** [comment]
Will move this to 0.9.0 for the missing schemas.

**Comment by jsancio** [comment]
We should make sure that we check JSON input for POST /v2/apps and PUT /v2/apps/{id} (when it is a new object) against the schema.

We had a user miss-spelled the `labels` as `lables` and he didn't get any errors. This kind of programming errors are hard to debug. Checking the input JSON against the schema will help us return nicer errors.

**Comment by ConnorDoyle** [comment]
In the interest of keeping things DRY, maybe we should also investigate automatic generation of JSON schemata for arbitrary `play.api.json.Reads` instances.

**Comment by kolloch** [comment]
*nitpick*

@ConnorDoyle I think you do not really propose JSON-schema generation for _arbitrary_ `Reads` because you would have to analyse arbitrary code.

**Comment by ConnorDoyle** [comment]
@kolloch yes, you're right of course.  The implementation could get complex really quickly.  I think the general goal is still valid though: somehow ensure the schema is in sync with parsers / validators.
